examples/aa-simple-dapp/src/hooks/useAsyncEffect.ts

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { useEffect, useMemo, useRef, useState } from "react";
+
+/**
+ * Hook to run an async effect on mount and another on unmount.
+ */
+export const useAsyncEffect = (
+  mountCallback: () => Promise<any>,
+  deps: any[] = [],
+  unmountCallback: () => Promise<any> = () => Promise.resolve()
+): UseAsyncEffectResult => {
+  const isMounted = useRef(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<unknown>(undefined);
+  const [result, setResult] = useState<any>();
+
+  useEffect(() => {
+    isMounted.current = true;
+    return () => {
+      isMounted.current = false;
+    };
+  }, []);
+
+  useEffect(() => {
```

Comment:
isn't this effect going to run every time the `deps` array updates? Should this no-op when `isMounted` is true?

Reply:
I ask because it seems unintended based on the description of the effect:
`Hook to run an async effect on mount and another on unmount`

Reply:
yes when deps update, it is intended to be run again, just like useeffect.